### ISSUE TITLE
Add Prefix dynamic parameter to Select-Object when ExpandProperty and Property are specified.

### ISSUE THREAD

**Issue by AjayKMehta** [PowerShell/PowerShell#12258]
# Summary

It would be great if an optional **Prefix** (dynamic?) parameter could be added to `Select-Object` when **ExpandProperty** and **Property** are specified to avoid errors when there are duplicate property names. 

# Problem

```powershell
Get-Module -ListAvailable | 
   select -Property Name -ExpandProperty RequiredModules
```

This results in an error:

> **Select-Object: The property cannot be processed because the property "Name" already exists.**

# Existing Solution

The workaround is simple:

```powershell
Get-Module -ListAvailable | 
   select -Property @{L = 'Module'; E = {$_.Name}} -ExpandProperty RequiredModules  |
   select Module, Name, Version
```

(**NOTE:** I added second `select` to reduce clutter in output)

**Output:**

```shell
Module                     Name              Version
------                     ----              -------
PowerShellGet              PackageManagement 1.4.4
Terminal-Icons             Configuration
```

This is fine IMHO if there is only 1 duplicate. It would be real annoying if there are several properties that have the same name.

# Proposed Solution

Add a **Prefix** parameter to `Select-Object`:

```powershell
Get-Module -ListAvailable | 
   select -Property Name -ExpandProperty RequiredModules -Property -Prefix 'Module'  |
   select ModuleName, Name, Version
```

**Output:**

```shell
ModuleName  Name  Version
------                ----      -------
PowerShellGet   PackageManagement 1.4.4
Terminal-Icons   Configuration
```

Adding a prefix has an added benefit that you can distinguish between properties resulting from **ExpandProperty** and **Property** parameters.

**Comment by scriptingstudio** [comment]
@AjayKMehta what do we do if \<prefix>\<propertyname> already exists?

**Comment by AjayKMehta** [comment]
@scriptingstudio Throw an exception

**Comment by scriptingstudio** [comment]
It is called recursion: prefix of prefix of prefix... exception of exception of exception...

**Comment by vexx32** [comment]
@scriptingstudio gonna have to explain what you mean there. 

@AjayKMehta I'm not very clear on what the expected behavior is. Are you suggesting the prefix only be applied in the case of a property name collision?

**Comment by scriptingstudio** [comment]
To avoid name conflict and terminating errors I would suggest switch `-NoClobber` to exclude duplicates as cmdlet `import-module` does.

**Comment by AjayKMehta** [comment]
@vexx32 No, the prefix should always be applied if specified. Throw an exception if **\<Prefix\>\<PropertyName\>** already exists.

**Comment by AjayKMehta** [comment]
`-NoClobber` helps to **exclude** duplicates but my stated problem is to **include** properties with the same name without error. Also, if you know *beforehand*, there is a property name conflict as there would be when dealing with a recursive data type, it would be much simpler to omit the property name from `-Property` parameter than specify `-NoClobber`.

If you want to include duplicates (which I do), `-NoClobber` doesn't change the status quo. 

Compare

```powershell
Get-Module -ListAvailable | 
   select -Property Name, Version -ExpandProperty RequiredModules -Prefix 'Module'  |
   select ModuleName, ModuleVersion, Name, Version
```

to

```powershell
Get-Module -ListAvailable | 
   select -Property @{L = 'ModuleName'; E = {$_.Name}}, @{L = 'ModuleVersion'; E = {$_.Version}} -ExpandProperty RequiredModules -NoClobber |
   select ModuleName, ModuleVersion, Name, Version
```

**Comment by AjayKMehta** [comment]
Can this be looked into?

**Comment by AjayKMehta** [comment]
This seems like a simple but useful fix. Can this be addressed in the near future?

**Comment by AjayKMehta** [comment]
This has been open for 5 years and is simple enough to address. Can this be looked into?

Thanks.

**Comment by AjayKMehta** [comment]
Maybe the prefix should be applied to properties coming from `-ExpandProperty` and this can be made more clear by naming the parameter `-ExpandedPrefix`?

**Comment by xtqqczze** [comment]
@iSazonov Please add label `KeepOpen`.

**Comment by AjayKMehta** [comment]
This has been open for over 5 years and is simple to address. Please keep this open.

Thanks.
